lib/cocoapods/external_sources/abstract_external_source.rb

```diff
@@ -111,11 +111,15 @@ def pre_download(sandbox)
         title = "Pre-downloading: `#{name}` #{description}"
         UI.titled_section(title,  :verbose_prefix => '-> ') do
           target = sandbox.pod_dir(name)
-          download_result = Downloader.download(download_request, target, :can_cache => can_cache)
+          begin
+            download_result = Downloader.download(download_request, target, :can_cache => can_cache)
+          rescue Pod::DSLError => e
+            raise Informative, "Failed to load '#{name}' podspec: #{e.message}"
```

Comment:
@benasher44 do you mean wrap the `sandbox.store_podspec(name, spec.to_pretty_json, true, true)` call with `begin` and `rescue` or do you mean line 123 `store_podspec(sandbox, spec)` to be wrapped?